### PR TITLE
sorter/leveldb(cdc): add leveldb-like interfaces

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve? 

Add leveldb-like interfaces, so that we can replace goleveldb with other databases easily.

### Check List 

Tests 

 - Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/sorter/leveldb/db/db.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+// DB is an interface of a leveldb-like database.
+type DB interface {
+	Snapshot() (Snapshot, error)
+	Batch(cap int) Batch
+	Close() error
+	CollectMetrics(captureAddr string, id int)
+}
+
+// A Batch is a sequence of Sets, Merges, Deletes, and/or DeleteRanges that are
```
Comment (hi-rustin) [inline cdc/sorter/leveldb/db/db.go:24]: I don't quite understand the comment. Can you explain it?

File: cdc/sorter/leveldb/db/leveldb.go [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+	"github.com/syndtr/goleveldb/leveldb/opt"
+	"github.com/syndtr/goleveldb/leveldb/util"
+	"go.uber.org/zap"
+)
+
+// OpenDB opens a leveldb.
+func OpenDB(ctx context.Context, id int, cfg *config.SorterConfig) (DB, error) {
+	lcfg := cfg.LevelDB
+	var option opt.Options
+	option.OpenFilesCacheCapacity = lcfg.MaxOpenFiles / lcfg.Count
```
Comment (hi-rustin) [inline cdc/sorter/leveldb/db/leveldb.go:39]: Perhaps it would be better to initialize these parameters by abstracting them into a function?

Comment (overvenus) [reply cdc/sorter/leveldb/db/leveldb.go:39]: Do you mean builder pattern? `Builder().SetA().SetB().Build()`

File: cdc/sorter/leveldb/db/db.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package db
```
Comment (amyangfei) [inline cdc/sorter/leveldb/db/db.go:14]: Better to move the interface definition to separated path, instead of the sub-path of leveldb

Comment (amyangfei) [reply cdc/sorter/leveldb/db/db.go:14]: Besides do we need a concrete package name, `db` is confused.

**Comment by asddongmen** [comment]
rest LGTM

File: cdc/sorter/leveldb/db/db.go (hunk 3 above)
Comment (overvenus) [reply cdc/sorter/leveldb/db/db.go:14]: Any suggestion about the name?

Comment (overvenus) [reply cdc/sorter/leveldb/db/db.go:14]: I'm going to move these files to pkg/db, and name it package db. Feel free to suggest a better name, thanks!

File: pkg/db/leveldb.go [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+	"github.com/syndtr/goleveldb/leveldb/iterator"
+	"github.com/syndtr/goleveldb/leveldb/opt"
+	"github.com/syndtr/goleveldb/leveldb/util"
+	"go.uber.org/zap"
+)
+
+// OpenDB opens a leveldb.
+func OpenDB(ctx context.Context, id int, path string, cfg *config.DBConfig) (DB, error) {
+	var option opt.Options
+	option.OpenFilesCacheCapacity = cfg.MaxOpenFiles / cfg.Count
```
Comment (hi-rustin) [inline pkg/db/leveldb.go:38]: What I mean is that the process of collecting and completing options can be a separate function.

**Review by hi-rustin — APPROVED** [review]
rest LGTM

**Review by asddongmen — APPROVED** [review]

**Pull request merged**